framework/test/unit/modules/token/api.spec.ts

```diff
@@ -0,0 +1,417 @@
+					Buffer.from([0, 0, 0, 1, 0, 0, 0, 0]),
+				),
+			).resolves.toEqual(defaultEscrowAmount);
+		});
+	});
+
+	describe('accountExists', () => {
+		it('should return true if account exist', async () => {
+			await expect(api.accountExists(apiContext, defaultAddress)).resolves.toBeTrue();
+		});
+
+		it('should return faslse if account does not exist', async () => {
+			await expect(api.accountExists(apiContext, getRandomBytes(20))).resolves.toBeFalse();
+		});
+	});
+
+	describe('getNextAvailableLocalID', () => {
+		it('should return next available local ID', async () => {
+			await expect(api.getNextAvailableLocalID(apiContext)).resolves.toEqual(
+				Buffer.from([0, 0, 0, 5]),
+			);
+		});
+	});
+
+	describe('initializeToken', () => {
+		it('should reject if supply already exist', async () => {
+			await expect(
+				api.initializeToken(apiContext, defaultTokenID.slice(CHAIN_ID_LENGTH)),
+			).rejects.toThrow('Token is already initialized');
+		});
+
+		it('should create empty supply', async () => {
+			await expect(
+				api.initializeToken(apiContext, defaultTokenID.slice(CHAIN_ID_LENGTH)),
+			).rejects.toThrow('Token is already initialized');
+		});
+
+		it('should not update next available local ID if local ID is less than existing one', async () => {
+			await expect(
+				api.initializeToken(apiContext, Buffer.from([0, 0, 0, 2])),
+			).resolves.toBeUndefined();
+			await expect(api.getNextAvailableLocalID(apiContext)).resolves.toEqual(
+				Buffer.from([0, 0, 0, 5]),
+			);
+		});
+
+		it('should update next available local ID if local ID is greater than existing one', async () => {
+			await expect(
+				api.initializeToken(apiContext, Buffer.from([0, 0, 0, 7])),
+			).resolves.toBeUndefined();
+			await expect(api.getNextAvailableLocalID(apiContext)).resolves.toEqual(
+				Buffer.from([0, 0, 0, 8]),
+			);
+		});
+	});
+
+	describe('mint', () => {
+		it('should reject if token is not native', async () => {
+			await expect(
+				api.mint(apiContext, defaultAddress, defaultForeginTokenID, BigInt(10000)),
+			).rejects.toThrow('Only native token can be minted');
+		});
+
+		it('should reject amount is less than zero', async () => {
+			await expect(
+				api.mint(apiContext, defaultAddress, defaultTokenID, BigInt(-1)),
+			).rejects.toThrow('Amount must be a positive integer to mint');
+		});
+
+		it('should reject if supply does not exist', async () => {
+			await expect(
+				api.mint(apiContext, defaultAddress, Buffer.from([0, 0, 0, 1, 0, 0, 0, 3]), BigInt(100)),
+			).rejects.toThrow('is not initialized to mint');
+		});
+
+		it('should reject if supply exceeds max uint64', async () => {
+			await expect(
+				api.mint(apiContext, defaultAddress, defaultTokenID, BigInt(2) ** BigInt(64)),
+			).rejects.toThrow('Supply cannot exceed MAX_UINT64');
+		});
+
+		it.todo('should reject if recipient does not exist and minBalance is not set');
+		it.todo('should reject if recipient does not exist and amount is less than minBalance');
```

Comment:
why todo?

Reply:
I left it todo for min balance check, because `minBalance` will be removed in favor of subtracting account creation fee from sender. I created one task to do after LIP has been updated